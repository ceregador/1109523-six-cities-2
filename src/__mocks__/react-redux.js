export const connectAdvanced = jest.fn().mockImplementation((component) => component);
const mockConnect = () => connectAdvanced;

export const connect = jest.fn().mockImplementation(mockConnect);
