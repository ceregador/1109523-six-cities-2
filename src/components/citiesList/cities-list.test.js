import React from 'react';
import {connect, connectAdvanced} from 'react-redux';
import CitiesList from './cities-list.jsx';
import City from '../city/city.jsx';
import Renderer from 'react-test-renderer';

jest.mock(`react-redux`);
jest.mock(`../city/city.jsx`);

it(`renders and connects correctly`, () => {
  const tree = Renderer
      .create(
          <CitiesList
            cities={[{name: ``, key: ``}]}
            activeCityName={``}
            onChangeCity={() => null}
            loadOffers={() => null}
          />)
      .toJSON();

  expect(connect).toHaveBeenCalledTimes(1);
  expect(connect).toHaveBeenCalledWith(null, expect.any(Object));

  const mapDispatchToProps = connect.mock.calls[0][1];
  expect(mapDispatchToProps).toHaveProperty(`loadOffers`);

  expect(connectAdvanced).toHaveBeenCalledWith(CitiesList);

  expect(City).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
