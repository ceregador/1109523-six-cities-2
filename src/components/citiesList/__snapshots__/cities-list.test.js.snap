// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and connects correctly 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  />
</section>
`;
